crates/swc_ecma_transforms_compat/src/es2022/class_properties/mod.rs

```diff
@@ -384,20 +394,17 @@ impl ClassProperties {
 
         let has_super = class.super_class.is_some();
 
-        let mut typescript_constructor_properties = vec![];
-
-        let mut constructor_exprs = vec![];
+        let mut constructor_inits = MemberInitMap::default();
         let mut vars = vec![];
-        let mut extra_stmts = vec![];
+        let mut extra_inits = MemberInitMap::default();
```

Comment:
Can you add some comment about the reason we use an indexed hashmap here? Is a vector not enough?

Reply:
added

Reply:
Thanks!